src/classes/PPlan_Primary_TDTM.cls

```diff
@@ -65,6 +85,18 @@ public with sharing class PPlan_Primary_TDTM extends TDTM_Runnable {
         }
         // BEFORE UPDATE
         if ( newlist != null && triggerAction == TDTM_Runnable.Action.BeforeUpdate && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.PPlan_Primary_TDTM_Before_Update)) {
+           for (integer i=0; i<newlist.size(); i++) {
+                Program_Plan__c newPPlan = (Program_Plan__c)newlist[i];
+                Program_Plan__c oldPPlan = (Program_Plan__c)oldlist[i];
+                if (newPPlan.Is_Primary__c == true && newPPlan.Is_Primary__c != oldPPlan.Is_Primary__c) {
+                    // Make Plan "Current" if "Is Primary" is checked.
+                    newPPlan.Status__c = STATUS_CURRENT_VAL;
+                }
+            }
+
+        }
+        // AFTER UPDATE
+        if ( newlist != null && triggerAction == TDTM_Runnable.Action.AfterUpdate && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.PPlan_Primary_TDTM_After_Update)) {
```

Comment:
Same as above. Why are we changing this logic to be performed on AfterUpdate instead BeforeUpdate, and whether we need to use DmlWrapper in this case.

Reply:
If it's true we needed to use DmlWrapper, why weren't the tests catching this?